src/explore-education-statistics-common/src/modules/find-statistics/components/charts/LineChartBlock.tsx

```diff
@@ -73,56 +70,56 @@ export class LineChartBlock extends Component<ChartProps> {
     }
 
     return (
-      <LineChart
-        width={900}
-        height={this.props.height || 300}
-        data={chartData}
-        margin={{ top: 5, right: 30, left: 20, bottom: 25 }}
-      >
-        <Tooltip content={CustomToolTip} />
-        {chartDataKeys.length > 1 ? (
-          <Legend verticalAlign="top" height={36} />
-        ) : (
-          ''
-        )}
-        <CartesianGrid strokeDasharray="3 3" />
-        <XAxis
-          dataKey="name"
-          label={{
-            offset: 5,
-            position: 'bottom',
-            value: xAxis.title,
-          }}
-          padding={{ left: 20, right: 20 }}
-          tickMargin={10}
-        />
-        <YAxis
-          label={{
-            angle: -90,
-            offset: 0,
-            position: 'left',
-            value: yAxis.title,
-          }}
-          scale="auto"
-          unit="%"
-          domain={yAxisDomain}
-        />
-        {chartDataKeys.map((dataKey, index) => (
-          <Line
-            key={index}
-            name={labels[dataKey]}
-            type="linear"
-            dataKey={dataKey}
-            stroke={colours[index]}
-            fill={colours[index]}
-            strokeWidth="5"
+      <ResponsiveContainer width={900} height={this.props.height || 300}>
+        <LineChart
+          data={chartData}
+          margin={{ top: 5, right: 30, left: 20, bottom: 25 }}
+        >
+          <Tooltip content={CustomToolTip} />
+          {chartDataKeys.length > 1 ? (
+            <Legend verticalAlign="top" height={36} />
+          ) : (
+            ''
+          )}
+          <CartesianGrid strokeDasharray="3 3" />
+          <XAxis
+            dataKey="name"
+            label={{
+              offset: 5,
+              position: 'bottom',
+              value: xAxis.title,
+            }}
+            padding={{ left: 20, right: 20 }}
+            tickMargin={10}
+          />
+          <YAxis
+            label={{
+              angle: -90,
+              offset: 0,
+              position: 'left',
+              value: yAxis.title,
+            }}
+            scale="auto"
             unit="%"
-            legendType={symbols[index]}
-            activeDot={{ r: 3 }}
-            dot={props => <Symbols {...props} type={symbols[index]} />}
+            domain={yAxisDomain}
           />
-        ))}
-      </LineChart>
+          {chartDataKeys.map((dataKey, index) => (
+            <Line
+              key={index}
```

Comment:
Can't we use the `dataKey` instead of the `index` here? 

It's generally bad practice to use indexes when you have something else to use for it (`dataKey` looks suitable).

Reply:
The datakey is something that could end up being set by the user, they may add the key in multiple times.

Reply:
The users can set the data key? As in the admin? Couldn't we validate it so that they have to provide unique data keys?

Reply:
It's a configurable value. They can add what they like to the charts. They want to show it twice, I see no reason to stop them.

Reply:
Okay, fair enough, happy with that for the time being